src/cart/crt_rpc.c

```diff
@@ -1628,8 +1628,11 @@ crt_rpc_priv_init(struct crt_rpc_priv *rpc_priv, crt_context_t crt_ctx, bool srv
 
 	crt_rpc_inout_buff_init(rpc_priv);
 
-	rpc_priv->crp_timeout_sec = (ctx->cc_timeout_sec == 0 ? crt_gdata.cg_timeout :
-				     ctx->cc_timeout_sec);
+	if (srv_flag)
+		rpc_priv->crp_timeout_sec = rpc_priv->crp_req_hdr.cch_src_timeout;
```

Comment:
this should be set only if cch_src_timeout != 0. older clients will not be setting this field, in which case we need to use default timeout (same logic as done for clients in else clause)